### PR TITLE
tools/functional-tester/etcd-agent: add status rpc

### PR DESCRIPTION
**Pull request by xiang90**
/cc @yichengq

### CODE REVIEW

File: tools/functional-tester/etcd-agent/agent.go [hunk 1]
```diff
@@ -21,9 +21,20 @@ import (
 	"os/exec"
 	"path"
 	"time"
+
+	"github.com/coreos/etcd/tools/functional-tester/etcd-agent/client"
+)
+
+const (
+	stateUninitialized = "uninitialized"
```
Comment (yichengq) [inline tools/functional-tester/etcd-agent/agent.go]: stateIdle?

File: tools/functional-tester/etcd-agent/agent.go [hunk 2]
```diff
@@ -43,7 +54,7 @@ func newAgent(etcd string) (*Agent, error) {
 		return nil, err
 	}
 
-	return &Agent{cmd: c, logfile: f}, nil
+	return &Agent{state: "uninitialized", cmd: c, logfile: f}, nil
```
Comment (yichengq) [inline tools/functional-tester/etcd-agent/agent.go]: do not use string directly

File: tools/functional-tester/etcd-agent/agent.go (hunk 1 above)
Comment (xiang90) [inline tools/functional-tester/etcd-agent/agent.go]: i like uninitialized much better

**Comment by xiang90** [comment]
@yichengq Fixed. PTAL.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**